src/Control/Retry.hs

```diff
@@ -157,7 +198,7 @@ fibonacciBackoff
     :: Int
     -- ^ Base delay in microseconds
     -> RetryPolicy
```

Comment:
I think this all works with the higher rank RetryPolicy change you suggested above.